src/mongo/mongoConnectionStrings.ts

```diff
@@ -3,19 +3,55 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-export function getDatabaseNameFromConnectionString(connectionString: string): string | undefined {
-    //  Connection strings follow the following format (https://docs.mongodb.com/manual/reference/connection-string/):
-    // mongodb://[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]][/[database][?options]]
+function parseConnectionString(connectionString: string): string[] | undefined {
+    // Connection strings follow the following format (https://docs.mongodb.com/manual/reference/connection-string/):
+    //   mongodb[+srv]://[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]][/[database][?options]]
     // Some example connection strings:
-    //   mongodb://dbuser:dbpassword@dbname.mlab.com:14118/dbname
-    //   mongodb://db1.example.net:27017,db2.example.net:2500/?replicaSet=test
-    //   mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/
+    //   mongodb://dbuser:dbpassword@dbname.mlab.com:14118
+    //   mongodb+srv://db1.example.net:27017,db2.example.net:2500/?replicaSet=test
+    //   mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/database?ssh=true
+    // Regex splits into five parts:
+    //   Full match
+    //   mongodb[+srv]
+    //   [username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]]
+    //   [database]
+    //   [?options]
+
+    try {
+        let matches = connectionString.match('(^mongodb(?:[+]srv)?):\/\/([^\/]*)(?:\/([^/?]*)(.*))?$');
+        if (matches && matches.length === 5) {
+            return matches.map(str => {
+                if (str === "") {
+                    str = undefined;
+                }
+                return str;
+            });
+        }
+    } catch (error) {
+        // Shouldn't happen, but ignore if does
+    }
+
+    return undefined;
+}
+
+export function getDatabaseNameFromConnectionString(connectionString: string): string | undefined {
+    try {
+        let [, , , database] = parseConnectionString(connectionString);
+        return database;
+    } catch (error) {
+        // Shouldn't happen, but ignore if does
+    }
+
+    return undefined;
+}
 
+export function addDatabaseToConnectionString(connectionString: string, databaseName: string): string | undefined {
     try {
-        let matches = connectionString.match(/[a-zA-Z]+:\/\/[^/]+(\/([^/?]+))/);
-        if (matches) {
-            let [, , database] = matches;
-            return database;
+        let [, , , parsedDatabaseName] = parseConnectionString(connectionString);
```

Comment:
Why do you need to get parsedDatabaseName?

Reply:
In case if connection string already contains database name I don't want to replace it, but if you think that it's better to replace I won't need it anymore. mongodb+srv://dbuser:dbpassword@dbname.mlab.com:14118/database`, 'notdatabase'

Reply:
This needs to show some example transformations in the comments.